composeApp/src/commonMain/kotlin/us/docbee/docbeeapp/domain/usecases/EmailSignupUseCase.kt

```diff
@@ -1,12 +1,83 @@
 import us.docbee.docbeeapp.domain.models.UserSignupResult
+import us.docbee.docbeeapp.domain.models.signup.SignUpParams
 import us.docbee.docbeeapp.domain.repositories.EmailAuthRepository
+import us.docbee.docbeeapp.utils.hasSpecialChars
+import us.docbee.docbeeapp.utils.isValidEmail
 
 class EmailSignupUseCase(
     private val emailAuthRepository: EmailAuthRepository
 ) {
-    suspend fun createUser(email: String, password: String): UserSignupResult {
-        return emailAuthRepository.signup(email, password)
+    suspend fun createUser(userInformation: SignUpParams): UserSignupResult {
+        val nameValidation = validateNames(userInformation.name)
+        val lastNameValidation = validateNames(userInformation.lastName)
+        val emailValidation = validateEmail(userInformation.email)
+        val dateOfBirthValidation = validateDateOfBirth(userInformation.dateOfBirth)
+        val phoneNumberValidation = validatePhoneNumber(userInformation.phoneNumber)
+        val passwordValidation = validatePassword(userInformation.password)
+
+        if (nameValidation !is SimpleTextValidation.Valid || lastNameValidation !is SimpleTextValidation.Valid
+            || emailValidation !is EmailTextValidation.Valid || dateOfBirthValidation !is DateOfBirthValidation.Valid
+            || phoneNumberValidation !is SimpleTextValidation.Valid || passwordValidation !is PasswordValidation.Valid
+        ) {
+            return UserSignupResult.SignupFormValidation(
+                name = nameValidation,
+                lastName = lastNameValidation,
+                email = emailValidation,
+                dateOfBirth = dateOfBirthValidation,
+                phoneNumber = phoneNumberValidation,
+                password = passwordValidation
+            )
+        }
+
+        return emailAuthRepository.signup(userInformation.email, userInformation.password)
+    }
+
+    private fun validateNames(name: String): SimpleTextValidation {
+        return if (name.isEmpty()) {
+            SimpleTextValidation.InvalidLength
+        } else {
+            SimpleTextValidation.Valid
+        }
+    }
+
+    private fun validateEmail(email: String): EmailTextValidation {
+        return when {
+            email.isEmpty() -> EmailTextValidation.EmptyValue
+            !isValidEmail(email) -> EmailTextValidation.InvalidEmail
+            else -> EmailTextValidation.Valid
+        }
+    }
+
+    private fun validateDateOfBirth(date: String): DateOfBirthValidation {
+        // TODO WIP validate age and date
+        return when {
+            date.isEmpty() -> DateOfBirthValidation.EmptyValue
+            else -> DateOfBirthValidation.Valid
+        }
+    }
+
+    private fun validatePhoneNumber(phone: String): SimpleTextValidation {
+        return when {
+            phone.isEmpty() -> SimpleTextValidation.EmptyValue
+            phone.length < 8 || phone.length > 13 -> SimpleTextValidation.InvalidLength
```

Comment:
Do you think that the 8 and 13 can be a constant to document, maybe a business rule?